src/WebManager/app/src/components/multi-tag-list.tsx

```diff
@@ -0,0 +1,330 @@
+﻿import * as React from "react";
+import { CancelTokenSource } from "axios";
+import { Docker } from "../services/docker";
+import { MultiManifest } from "./multimanifest";
+import {
+    Button,
+    ButtonType
+} from "office-ui-fabric-react/lib/Button";
+
+export interface IMultiTagListProps {
+    service: Docker,
+    params: any,
+    onLoadFailure?: (err: any) => void
+}
+
+interface IMultiTagListState {
+    tags: string[],
+    hasMoreTags: boolean,
+    error: string,
+    checkedTags: string[],
+    multiManifestTags: string[],
+    targetTag: string
+}
+
+export class MultiTagList extends React.Component<IMultiTagListProps, IMultiTagListState> {
+    private cancel: CancelTokenSource = null;
+    private hash: { [key: string]: string } = {};
+   
+    constructor(props: IMultiTagListProps) {
+        super(props);
+        this.state = {
+            tags: null,
+            hasMoreTags: true,
+            error: null,
+            checkedTags: [],
+            multiManifestTags: [],
+            targetTag: null
+        };
+    }
+
+    changeToggle(e: React.FormEvent<HTMLInputElement>) {
+
+        let tempArray = this.state.checkedTags;
+        if (e.target.checked === true) {
+            if (this.followsConvention(e.target.value) && this.hash[e.target.value] != "") {
```

Comment:
In this case I think it would be better to use !== because we know that the dictionary has the type string for the values